lib/WUI/nhttp/send_json.cpp

```diff
@@ -0,0 +1,124 @@
+#include "send_json.h"
+#include "handler.h"
+#include "headers.h"
+
+#include <timing.h>
+#include <segmented_json_macros.h>
+#include <http/chunked.h>
+#include <transfers/monitor.hpp>
+
+using transfers::Monitor;
+using namespace json;
+using namespace http;
+
+namespace nhttp::handler {
+
+JsonResult TransferRenderer::renderState(size_t resume_point, json::JsonOutput &output, TransferState &state) const {
+    auto transfer_status = Monitor::instance.status(resume_point != 0);
+    // Keep the indentation of the JSON in here!
+    // clang-format off
+    JSON_START;
+        JSON_OBJ_START
+            if (transfer_status.has_value()) {
+                if (resume_point != 0 && transfer_status->id != state.transfer_id)
+                    // if transfer changes mid report, bail out
+                    transfer_status.reset();
+                else
+                    state.transfer_id = transfer_status->id;
```

Comment:
I'm not sure we (the team) explicitly agreed on this one, but… I tend to use the block braces `{ }` even with single-line blocks around if-else.

The reason is, it is somewhat easy to accidentally add another (indented) line into the „block“, but it would not be conditional. This is somewhat alleviated by the fact that the automatic formatter would reformat it one level up again, so the reviewer would have less chance to get confused, but I think the precaution is worth having two more characters in the source code 😇

I think I might point this precaution out in the next retrospective